quine/src/main/scala/com/thatdot/quine/app/ingest/KafkaSrcDef.scala

```diff
@@ -56,18 +63,38 @@ object KafkaSrcDef {
 
     val keyDeserializer: ByteArrayDeserializer = new ByteArrayDeserializer() //NO-OP
 
-    ConsumerSettings(graph.system, keyDeserializer, deserializer)
-      .withBootstrapServers(bootstrapServers)
-      .withGroupId(groupId)
-      // Note: The ConsumerSettings stop-timeout delays stopping the Kafka Consumer
-      // and the stream, but when using drainAndShutdown that delay is not required and can be set to zero (as below).
-      // https://doc.akka.io/docs/alpakka-kafka/current/consumer.html#draining-control
-      // We're calling .drainAndShutdown on the Kafka [[Consumer.Control]]
-      .withStopTimeout(Duration.Zero)
-      .withProperties(
-        AUTO_OFFSET_RESET_CONFIG -> autoOffsetReset.name,
-        SECURITY_PROTOCOL_CONFIG -> securityProtocol.name
-      )
+    // Configure consumer with JAAS config if exists
+    saslAuthentication match {
+      case Some(KafkaSaslAuthentication.Plain(jaasConfig, saslMechanism)) => { 
+        ConsumerSettings(graph.system, keyDeserializer, deserializer)
+          .withBootstrapServers(bootstrapServers)
+          .withGroupId(groupId)
+          // Note: The ConsumerSettings stop-timeout delays stopping the Kafka Consumer
+          // and the stream, but when using drainAndShutdown that delay is not required and can be set to zero (as below).
+          // https://doc.akka.io/docs/alpakka-kafka/current/consumer.html#draining-control
+          // We're calling .drainAndShutdown on the Kafka [[Consumer.Control]]
+          .withStopTimeout(Duration.Zero)
+          .withProperties(
+            "sasl.jaas.config" -> jaasConfig,
+            "sasl.mechanism" -> saslMechanism,
+            // "sasl.kerberos.service.name" -> "kafka",
+            AUTO_OFFSET_RESET_CONFIG -> autoOffsetReset.name,
+            SECURITY_PROTOCOL_CONFIG -> securityProtocol.name
+          )}
+      case None => {
+        ConsumerSettings(graph.system, keyDeserializer, deserializer)
+          .withBootstrapServers(bootstrapServers)
+          .withGroupId(groupId)
+          // Note: The ConsumerSettings stop-timeout delays stopping the Kafka Consumer
+          // and the stream, but when using drainAndShutdown that delay is not required and can be set to zero (as below).
+          // https://doc.akka.io/docs/alpakka-kafka/current/consumer.html#draining-control
+          // We're calling .drainAndShutdown on the Kafka [[Consumer.Control]]
+          .withStopTimeout(Duration.Zero)
+          .withProperties(
+            AUTO_OFFSET_RESET_CONFIG -> autoOffsetReset.name,
+            SECURITY_PROTOCOL_CONFIG -> securityProtocol.name
+          )}
```

Comment:
not super happy with this pattern but was following what was done with `offsetCommitting`

Reply:
each of `offsetCommitting`'s `case`s is a single line -- since this is multiple lines, a simpler pattern is appropriate.

Here's an example, although you could just as well use an Option.fold or if statements if you prefer.
```
val consumerWithoutSasl = ConsumerSettings(graph.system, keyDeserializer, deserializer)
  .withBootstrapServers(bootstrapServers)
  .withGroupId(groupId)
  // Note: The ConsumerSettings stop-timeout delays stopping the Kafka Consumer
  // and the stream, but when using drainAndShutdown that delay is not required and can be set to zero (as below).
  // https://doc.akka.io/docs/alpakka-kafka/current/consumer.html#draining-control
  // We're calling .drainAndShutdown on the Kafka [[Consumer.Control]]
  .withStopTimeout(Duration.Zero)
  .withProperties(
    AUTO_OFFSET_RESET_CONFIG -> autoOffsetReset.name,
    SECURITY_PROTOCOL_CONFIG -> securityProtocol.name
  )}

saslAuthentication match {
  case Some(KafkaSaslAuthentication.Plain(jaasConfig, saslMechanism)) =>
    consumerWithoutSasl.withProperties(
      "sasl.jaas.config" -> jaasConfig,
      "sasl.mechanism" -> saslMechanism
    )
  case None =>
    consumerWithoutSasl
}
```

Reply:
Or build the properties map seperately, or, or, or.... 😁 Whatever pattern you are more happy with.

Reply:
I agree, that is a much cleaner approach, dont know why I didnt do that in the first place. Ill add something along those lines. Thanks